arduino-ide-extension/src/browser/dialogs/ide-updater/ide-updater-dialog.tsx

```diff
@@ -117,8 +57,23 @@ export class IDEUpdaterDialogProps extends DialogProps {}
 
 @injectable()
 export class IDEUpdaterDialog extends AbstractDialog<UpdateInfo> {
+  private onError: Disposable;
+  private onDownloadProgressChanged: Disposable;
+  onDownloadFinished: Disposable;
```

Comment:
missing `private` and how the (`Disposable`) fields are initialized is quite complicated.

Why not something like:

```ts
class IDEUpdaterDialog {

  constructor() {
  // current code
  }
  
  protected void postConstruct(): void {
    this.onError = ...;
    this.onDownloadFinished = ...;
    this.onDownloadProgressChanged = ...;
  }

  private init(): void {
  // current code without the if (!this.onError) and if (!this.onDownloadFinished), etc.
  }

 Maybe you can rename the `init` to `resetDownloadProgress,` or it's up to you. 


}
```

Please favor `onDid[Object]Did[Verb]` pattern: `onDownloadDidFinish`, `onDownloadProgressDidChange`.

Reply:
It totally makes sense, all of it. Thanks for taking the time to make this kind of suggestions!